frontend/catalyst/cuda/catalyst_to_cuda_interpreter.py

```diff
@@ -0,0 +1,851 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module implements a custom Jaxpr interpreter.
+
+  https://jax.readthedocs.io/en/latest/notebooks/Writing_custom_interpreters_in_Jax.html
+
+This custom Jaxpr interpreter will interpret Jaxpr code generated from the tracing
+of catalyst programs and it will interpret it in such a way that instead of interpreting
+Catalyst operations, it will issue calls to cuda-quantum operations.
+
+This effectively transforms a catalyst program into something that can generate cuda
+quantum kernels.
+
+This module also uses the CUDA-quantum API. Here is the reference:
+  https://nvidia.github.io/cuda-quantum/latest/api/languages/python_api.html
+"""
+
+import functools
+import json
+import operator
+from functools import reduce, wraps
+from typing import List
+
+import cudaq
+import jax
+import pennylane as qml
+from jax._src.util import safe_map
+from jax.tree_util import tree_unflatten
+
+from catalyst.jax_primitives import (
+    AbstractObs,
+    adjoint_p,
+    compbasis_p,
+    cond_p,
+    counts_p,
+    expval_p,
+    for_p,
+    func_p,
+    grad_p,
+    hamiltonian_p,
+    hermitian_p,
+    jvp_p,
+    namedobs_p,
+    print_p,
+    probs_p,
+    qalloc_p,
+    qdealloc_p,
+    qdevice_p,
+    qextract_p,
+    qinsert_p,
+    qinst_p,
+    qmeasure_p,
+    qunitary_p,
+    sample_p,
+    state_p,
+    tensorobs_p,
+    var_p,
+    vjp_p,
+    while_p,
+    zne_p,
+)
+from catalyst.jax_tracer import trace_to_jaxpr
+from catalyst.pennylane_extensions import QFunc
+from catalyst.utils.jax_extras import remove_host_context
+from catalyst.utils.patching import Patcher
+
+from .primitives import (
+    AbsCudaKernel,
+    AbsCudaQbit,
+    AbsCudaQReg,
+    AbsCudaQState,
+    AbsCudaSampleResult,
+    AbsCudaSpinOperator,
+    AbsCudaValue,
+    cuda_inst,
+    cudaq_adjoint,
+    cudaq_counts,
+    cudaq_expectation,
+    cudaq_getstate,
+    cudaq_make_kernel,
+    cudaq_observe,
+    cudaq_sample,
+    cudaq_spin,
+    kernel_qalloc,
+    mz_call,
+    qreg_getitem,
+)
+
+# We disable protected access in particular to avoid warnings with
+# cudaq._pycuda.
+# pylint: disable=protected-access
+# We also disable line-too-long because of a URL
+# pylint: disable=line-too-long
```

Comment:
We should be able to disable warnings on a per-line basis using e.g.
```python
# pylint: disable-next=line-too-long
```
Alternatively, the [`--ignore-long-lines`](https://pylint.pycqa.org/en/latest/user_guide/configuration/all-options.html#ignore-long-lines) flag can be used to have URLs be ignored by this check.

Reply:
Thanks! I didn't know there was a `disable-next`.